src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/OpenSslCrlCache.cs

```diff
@@ -268,13 +293,21 @@ private static string GetCachedCrlPath(string localFileName, bool mkDir=false)
                     {
                         foreach (GeneralNameAsn name in distributionPoint.DistributionPoint.Value.FullName)
                         {
-                            if (name.Uri != null &&
-                                Uri.TryCreate(name.Uri, UriKind.Absolute, out Uri? uri) &&
-                                uri.Scheme == "http")
+                            if (name.Uri != null)
                             {
-                                return name.Uri;
+                                if (Uri.TryCreate(name.Uri, UriKind.Absolute, out Uri? uri) &&
+                                    uri.Scheme == "http")
+                                {
+                                    return name.Uri;
+                                }
+                                else
+                                {
+                                    OpenSslX509ChainEventSource.Log.NonHttpCdpEntry(name.Uri);
+                                }
                             }
                         }
+
+                        OpenSslX509ChainEventSource.Log.NoMatchingCdpEntry();
```

Comment:
And here I thought that was an artifact of older patterns.

So the correct style, for dotnet/runtime, at least, is an IsEnabled check on both the outside and the inside of the method?   (I feel like EventSource's documentation said to have an IsEnabled check for each method, presumably to avoid things like creating the params array when it doesn't matter)  Or do we do them outside-only?

Reply:
Yep, linker will have right now difficulties propagating this information. The best is to call directly [methods](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.Shared.xml#L3-L8) that are substituted when event source is disabled